image/manifest.go

```diff
@@ -240,3 +240,34 @@ loop:
 	}
 	return nil
 }
+
+// ValidateManifestMediaType validate the manifest schema media-type
+// and its descendants fields media type, such as config and layers,
+// to make sure they match to spec definition, or returns an error if
+// the validation failed.
+func ValidateManifestMediaType(r io.ReadCloser) error {
+	header := v1.Manifest{}
+	defer r.Close()
+
+	buf, err := ioutil.ReadAll(r)
+	if err != nil {
+		return errors.Wrapf(err, "error reading the io stream")
+	}
+
+	err = json.NewDecoder(bytes.NewReader(buf)).Decode(&header)
+	if err != nil {
+		return errors.Wrap(err, "manifest format mismatch")
+	}
```

Comment:
You want an error-checked call to `r.Close()` here.  No need to keep the reader open after you've read it while you walk the descendants.